adapters/RPGRegionsAdapter/src/main/kotlin/com/caleb/typewriter/rpgregions/entries/action/DiscoverRegionActionEntry.kt

```diff
@@ -0,0 +1,41 @@
+package com.caleb.typewriter.griefdefender.entries.action
+
+import me.gabber235.typewriter.adapters.Colors
+import me.gabber235.typewriter.adapters.Entry
+import me.gabber235.typewriter.adapters.modifiers.Help
+import me.gabber235.typewriter.entry.Criteria
+import me.gabber235.typewriter.entry.Modifier
+import me.gabber235.typewriter.entry.entries.ActionEntry
+import me.gabber235.typewriter.utils.Icons
+import net.islandearth.rpgregions.api.RPGRegionsAPI
+import net.islandearth.rpgregions.managers.data.region.WorldDiscovery
+import org.bukkit.entity.Player
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+@Entry("discover_region", "Create a discover for a RPGRegions region", Colors.RED, Icons.ADDRESS_BOOK)
+class DiscoverRegionActionEntry(
+    override val id: String = "",
+    override val name: String = "",
+    override val criteria: List<Criteria> = emptyList(),
+    override val modifiers: List<Modifier> = emptyList(),
+    override val triggers: List<String> = emptyList(),
+    @Help("The region to discover.")
+    private val region: String = "",
+) : ActionEntry {
+    override fun execute(player: Player) {
+        super.execute(player)
+        val region = RPGRegionsAPI.getAPI().managers.regionsCache.getConfiguredRegion(region) ?: return
```

Comment:
It feels incorrect that you query the cache. Normally caches are volatile and not the full list of items (only the ones that are used most often). Could you check with the developers of `RPGRegions` if this is the correct way to do it?

Reply:
This also explains why we should remove the cache after making changes.